apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -806,55 +806,67 @@ func (m *ModelManagerAPI) ListModelSummaries(ctx context.Context, req params.Mod
 		return result, errors.Trace(err)
 	}
 
-	modelInfos, err := m.state.ModelSummariesForUser(userTag, req.All && m.isAdmin)
+	var modelInfos []model.ModelSummary
+	if req.All {
+		if !m.isAdmin {
+			return result, fmt.Errorf("%w: cannot list all models as non-admin user", apiservererrors.ErrPerm)
+		}
+		modelInfos, err = m.modelService.ListAllModelSummaries(ctx)
```

Comment:
Seems like we potentially have a bug or logic problem here. When we look at `ModelSummary.UserAccess` it says `// UserAccess is model access level for the  current user.`. We don't supply what the current user is for this call so I am assuming the comment is wrong as it is scoped off to the owner?

I think the logic needs to be defined and either the comment updated or this service func needs to process access level with respect to the calling user.

Reply:
So when I did a test on 3.6 for this, the access was blank for the case of the `--all` flag. I have implemented the model summary to only include the user access level (and last login time) for the case where we are getting the users model info, if they provide the `--all` flag these fields are blank.

What are your opinions on that? I can update the comment to reflect this, or change the service func

Reply:
Thanks for the investigation. It might be worth double checking my thoughts below with @SimonRichardson or @manadart for sanity.

I think for the facade if this is how it currently works then that is fine. It is locked in behaviour that we should probably just document on the func it self.

As for the service calls I think they should be returning different structs types to represents the information better. We can use composition of the types to extend the user call with access levels.

I am personally against ambiguity in data returns where things may or may not be filled out based on conceptual context.